storage/s3.go

```diff
@@ -587,9 +595,28 @@ func (s *S3) MakeBucket(ctx context.Context, name string) error {
 	return err
 }
 
+// Session holds session.Session according to s3Opts
+// and it synchronizes access/modification.
+type Session struct {
```

Comment:
Can we make this type unexported since it is only used internally and not called by another package?